app/plugin.go

```diff
@@ -450,3 +451,98 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 
 	return nil
 }
+
+// GetPluginPublicKeys returns all public keys listed in the config.
+func (a *App) GetPluginPublicKeys() ([]*model.PublicKeyDescription, *model.AppError) {
+	config := a.Config().PluginSettings
+
+	return config.PublicKeys, nil
+}
+
+// GetPublicKey will return the actual public key saved in the `filename` file.
+func (a *App) GetPublicKey(filename string) ([]byte, *model.AppError) {
+	data, err := a.Srv.configStore.GetFile(filename)
+	if err != nil {
+		return nil, model.NewAppError("GetPublicKey", "api.plugin.get_public_key.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return data, nil
+}
+
+func (a *App) writePublicKeyFile(file string) *model.AppError {
+	_, filename := filepath.Split(file)
+	fileReader, err := os.Open(file)
+	if err != nil {
+		return model.NewAppError("AddPublicKey", "api.plugin.add_public_key.open.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	defer fileReader.Close()
+
+	data, err := ioutil.ReadAll(fileReader)
+	if err != nil {
+		return model.NewAppError("AddPublicKey", "api.plugin.add_public_key.read.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	err = a.Srv.configStore.SetFile(filename, data)
```

Comment:
Ugh, I can't believe I made `SetFile` accept `[]byte` instead of an `io.Reader`! Sorry about that :/

As an aside, one potential concern here is the clash between SAML certificate filenames and plugin signing key filenames. Should we require the file not to exist first (and thus be deleted intentionally)?

Reply:
Good point about the saml files. I think user should be able to update file with the same name. May be we can force the .asc file extension so our files will not collide with samls?

Reply:
Forcing an extension seems like a decent compromise.

Reply:
yep `.plugin.asc` is better